### ISSUE TITLE
TypeScript should error on static attribute called 'name'

### ISSUE THREAD

**Issue by vivainio** [Microsoft/TypeScript#4143]
"name" property on a function has a special meaning in JavaScript

(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name)

If you do

```typescript
class MyC {
 static name = "foo";
 foo() { 
  console.log(MyC.name);
 }
}

f = new MyC();
f.foo();
```

You get code that behaves unxpectedly (prints "MyC" instead of "foo". With minifaction, it will print a minified function name (e.g. "i").

Desired behavior: TypeScript should raise error or warning when you try to create static attribute called "name".

**Comment by mhegazy** [comment]
looks like a dupe of  #442
